tests/Doctrine/Migrations/Tests/Tools/Console/Command/DoctrineCommandTest.php

```diff
@@ -73,4 +74,27 @@ protected function execute(InputInterface $input, OutputInterface $output): int
             ['interactive' => false]
         );
     }
+
+    public function testDependencyFactoryIsSetFirst(): void
+    {
+        $exceptionThrown = false;
+
+        $dependencyFactory = $this->createMock(DependencyFactory::class);
+
+        try {
+            $command = new class ($dependencyFactory) extends DoctrineCommand
+            {
+                protected function configure(): void
+                {
+                    $this->getDependencyFactory();
+                }
+            };
+
+            self::assertFalse($command->getDefinition()->hasOption('db-configuration'));
+        } catch (DependenciesNotSatisfied $exception) {
+            $exceptionThrown = true;
+        }
```

Comment:
What I meant in https://github.com/doctrine/migrations/pull/1043/files#r484106894

```suggestion
        $command = new class ($dependencyFactory) extends DoctrineCommand
        {
            protected function configure(): void
            {
                $this->getDependencyFactory();
            }
        };

        self::assertFalse($command->getDefinition()->hasOption('db-configuration'));
```